pkg/config/config_template.yaml

```diff
@@ -1222,10 +1222,21 @@ api_key:
   #
   # connection_limit: 2000
 
+  ## @param compute_stats_by_span_kind - bool - default: false
+  ## @env DD_APM_COMPUTE_STATS_BY_SPAN_KIND - bool - default: false
+  ## Enables an additional stats computation check on spans to see they have an eligible `span.kind` (server, consumer, client, producer).
+  ## If enabled, a span with an eligible `span.kind` will have stats computed. If disabled, only top-level and measured spans will have stats computed.
+  # compute_stats_by_span_kind: false
+
+  ## @param peer_service_stats_aggregation - bool - default: false
+  ## @env DD_APM_PEER_SERVICE_STATS_AGGREGATION - bool - default: false
+  ## Enables peer.service stats aggregation in the agent. If disabled, aggregated stats will not include peer.service as a dimension.
+  # peer_service_stats_aggregation: false
```

Comment:
The PR description says _This is a somewhat complementary but arguably necessary_ about `compute_stats_by_span_kind`. What's the impact of enabling one of the 2 new options without the other?
If understand correctly `compute_stats_by_span_kind` should be also set to `true` if `peer_service_stats_aggregation` is enabled, this can be error prone! Does it make sense to have a single option `peer_service_stats_aggregation` and always compute by span kind if it's enabled? (I might be missing context here).

Reply:
This is a great question. Let me explain:
- For Datadog tracers, most of the client spans where `peer.service` would be set are already marked as measured, so checking `span.kind` is unnecessary to get stats. So you don't _have_ to set `compute_stats_by_span_kind` in all cases.
- However, there are cases where `span.kind` is set to client/producer and the span is _not_ measured. Here's where you'd want both flags to be safe. If someone is submitting OTel traces, they definitely would want both options on.
- At the same time these cases might also be high in cardinality, so enabling `span.kind` computation could be unexpectedly expensive.
- Lastly, we may have edge cases where we have non-top-level server/consumer spans that we want to compute stats for. These spans won't have a `peer.service`.

Ultimately I think we want to get to a point where both behaviors are on by default. Having the options be separated and off by default lets us test in a safer, more granular fashion.

I'm wondering though about the logic you suggested. Maybe there's a way to get a little more precise here.

Reply:
Thank you for explaining! Let's capture all that in the documentation of these field.